sanic/mixins/routes.py

```diff
@@ -121,8 +121,11 @@ def decorator(handler):
                         "Expected either string or Iterable of host strings, "
                         "not %s" % host
                     )
-
-            if isinstance(subprotocols, (list, tuple, set)):
+            if isinstance(subprotocols, list):
+                # Ordered subprotocols, maintain order
+                subprotocols = tuple(subprotocols)
+            if isinstance(subprotocols, set):
```

Comment:
Can we make this `elif` so lists will not trigger it?

Reply:
Yes. I think this is a typo, should be `elif`.